subprojects/core/src/test/groovy/org/gradle/api/internal/project/taskfactory/AnnotationProcessingTasks.java

```diff
@@ -49,6 +50,7 @@ public class AnnotationProcessingTasks {
     public static class TestTask extends DefaultTask {
```

Comment:
This whole thing needs to be rewritten because you can't actually write tasks like this in real builds.  e.g., you can't pass null to the constructor like we're doing in all of these tests.